File: benchmarks/vm/16_numeric_map_mutation.lua
local t0 = os.clock()
local m = {}
for i = 0, 59999 do
    m[i] = i * 3
end

local total = 0
local hits = 0
for j = 0, 59999 do
    if m[j] ~= nil then
        hits = hits + 1
        total = total + m[j]
    end
    if (j % 4) == 0 then
        m[j] = nil
    end
end

local survivors = 0
for k = 0, 59999 do
    if m[k] ~= nil then
        survivors = survivors + 1
    end
end

local count = survivors
local score = total + (hits * 7) + (survivors * 11) + count
local total_ms = (os.clock() - t0) * 1000
print("count = " .. tostring(count))
print("score = " .. tostring(score))
print(string.format("BENCH_RESULT: %.3f ms", total_ms))

File: benchmarks/vm/16_numeric_map_mutation.rb
t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
m = {}
60_000.times do |i|
  m[i] = i * 3
end

total = 0
hits = 0
60_000.times do |j|
  if m.key?(j)
    hits += 1
    total += m.fetch(j, 0)
  end
  m.delete(j) if (j % 4) == 0
end

survivors = 0
60_000.times do |k|
  survivors += 1 if m.key?(k)
end

count = m.length
score = total + (hits * 7) + (survivors * 11) + count
total_ms = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - t0) * 1000
puts "count = #{count}"
puts "score = #{score}"
puts format("BENCH_RESULT: %.3f ms", total_ms)
